docs/docs/contributions/development/setting-up-pants.mdx

```diff
@@ -61,32 +61,30 @@ To free up space, run `rm -rf src/rust/engine/target`.
 Warning: this will cause Rust to recompile everything.
 :::
 
-## Step 3: (Optional) Set up a Git hook
+## Step 3: Set up a pre-push Git Hook
 
-We have a [Git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) that runs some useful checks and lints when you `git commit`.
+We have a [Git Hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) that runs some useful checks and lints when you `git push`. Running this locally can prevent easily avoidable CI failures such as whitespace or linting issues.
 
 To install this, run:
 
 ```bash
 $ build-support/bin/setup.sh
 ```
 
-If you commit frequently as part of your workflow you may find it annoying to have these run every time, so installing this hook is not required.
```

Comment:
Removing "Optional".

The frequency of run is now substantially lower, since it's on push instead of commit.